### ISSUE TITLE
Traefik 2.2.2 -> "Internal Server Error" (when IP whitelisting is enabled)

### ISSUE THREAD

**Issue by RogerSik** [containous/traefik#7014]
With the Traefik release 2.2.2 and the installation on our server we get now "Internal Server Error". Our fix was to revert back to 2.2.1.

The Traefik logs are now spamming following messages:

```
time="2020-07-09T08:18:07+02:00" level=error msg="Recovered from panic in HTTP handler [91.249.156.74:36968 - /favicon.ico]: runtime error: invalid memory address or nil pointer dereference" middlewareName=traefik-internal-recovery middlewareType=Recovery
time="2020-07-09T08:18:07+02:00" level=error msg="Stack: goroutine 92 [running]:\ngithub.com/containous/traefik/v2/pkg/middlewares/recovery.recoverFunc(0x2e33c40, 0xc0002e0660, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/recovery/recovery.go:47 +0x1db\npanic(0x242c760, 0x477fb60)\n\t/usr/local/go/src/runtime/panic.go:969 +0x166\ngithub.com/containous/traefik/v2/pkg/responsemodifiers.(*Builder).Build.func1(0xc0000ee120, 0x2, 0x0)\n\t/go/src/github.com/containous/traefik/pkg/responsemodifiers/response_modifier.go:55 +0x6d\nnet/http/httputil.(*ReverseProxy).modifyResponse(0xc000530230, 0x2e18100, 0xc0005ce9c0, 0xc0000ee120, 0xc0000fcf00, 0x0)\n\t/usr/local/go/src/net/http/httputil/reverseproxy.go:177 +0x41\nnet/http/httputil.(*ReverseProxy).ServeHTTP(0xc000530230, 0x2e18100, 0xc0005ce9c0, 0xc0000fce00)\n\t/usr/local/go/src/net/http/httputil/reverseproxy.go:278 +0x3e7\ngithub.com/containous/traefik/v2/pkg/middlewares/pipelining.(*pipelining).ServeHTTP(0xc0005cf240, 0x2e2a100, 0xc0005cc050, 0xc0000fce00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/pipelining/pipelining.go:36 +0xe7\ngithub.com/containous/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc000263bc0, 0x2e2a100, 0xc0005cc050, 0xc0000fce00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/accesslog/field_middleware.go:29 +0x18d\ngithub.com/vulcand/oxy/roundrobin.(*RoundRobin).ServeHTTP(0xc0000235e0, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)\n\t/go/pkg/mod/github.com/vulcand/oxy@v1.1.0/roundrobin/rr.go:147 +0x13d\ngithub.com/containous/traefik/v2/pkg/middlewares/emptybackendhandler.(*emptyBackend).ServeHTTP(0xc0005cf270, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/emptybackendhandler/empty_backend_handler.go:31 +0x168\ngithub.com/containous/traefik/v2/pkg/middlewares/tracing.(*forwarderMiddleware).ServeHTTP(0xc0002e0d20, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/tracing/forwarder.go:38 +0x609\ngithub.com/containous/traefik/v2/pkg/middlewares/ipwhitelist.(*ipWhiteLister).ServeHTTP(0xc000263c80, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/ipwhitelist/ip_whitelist.go:76 +0x3da\ngithub.com/containous/traefik/v2/pkg/middlewares/tracing.(*Wrapper).ServeHTTP(0xc0002e18f0, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/tracing/wrapper.go:57 +0x16a\ngithub.com/containous/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc000263cc0, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/accesslog/field_middleware.go:29 +0x18d\ngithub.com/gorilla/mux.(*Router).ServeHTTP(0xc00007fc20, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)\n\t/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x112\ngithub.com/containous/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0xc00053ca00, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/recovery/recovery.go:33 +0x104\ngithub.com/containous/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc000263ec0, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/accesslog/field_middleware.go:29 +0x18d\ngithub.com/containous/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0xc00011cdd8, 0x2e2a100, 0xc0005cc050, 0xc0000fc800, 0xc0005f8c70)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x37d\ngithub.com/containous/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1(0x2e2a100, 0xc0005cc050, 0xc0000fc800)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x83\nnet/http.HandlerFunc.ServeHTTP(0xc00053cae0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)\n\t/usr/local/go/src/net/http/server.go:2012 +0x44\ngithub.com/containous/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0xc00011cdc0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/handler_switcher.go:23
+0x70\ngithub.com/containous/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0xc00013f680, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)\n\t/go/src/github.com/containous/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:173 +0x10b\nnet/http.serverHandler.ServeHTTP(0xc0002241c0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)\n\t/usr/local/go/src/net/http/server.go:2807 +0xa3\nnet/http.initALPNRequest.ServeHTTP(0x2e33c40, 0xc0004818f0, 0xc0000f1500, 0xc0002241c0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)\n\t/usr/local/go/src/net/http/server.go:3381 +0x8d\nnet/http.(*http2serverConn).runHandler(0xc000502780, 0xc0005cc050, 0xc0000fc800, 0xc000188040)\n\t/usr/local/go/src/net/http/h2_bundle.go:5720 +0x8b\ncreated by net/http.(*http2serverConn).processHeaders\n\t/usr/local/go/src/net/http/h2_bundle.go:5454 +0x4e1\n" middlewareName=traefik-internal-recovery middlewareType=Recovery
```

```
Recovered from panic in HTTP handler [91.249.156.74:36968 - /favicon.ico]: runtime error: invalid memory address or nil pointer dereference
Stack: goroutine 92 [running]:
github.com/containous/traefik/v2/pkg/middlewares/recovery.recoverFunc(0x2e33c40, 0xc0002e0660, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)
	/go/src/github.com/containous/traefik/pkg/middlewares/recovery/recovery.go:47 +0x1db
panic(0x242c760, 0x477fb60)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/containous/traefik/v2/pkg/responsemodifiers.(*Builder).Build.func1(0xc0000ee120, 0x2, 0x0)
	/go/src/github.com/containous/traefik/pkg/responsemodifiers/response_modifier.go:55 +0x6d
net/http/httputil.(*ReverseProxy).modifyResponse(0xc000530230, 0x2e18100, 0xc0005ce9c0, 0xc0000ee120, 0xc0000fcf00, 0x0)
	/usr/local/go/src/net/http/httputil/reverseproxy.go:177 +0x41
net/http/httputil.(*ReverseProxy).ServeHTTP(0xc000530230, 0x2e18100, 0xc0005ce9c0, 0xc0000fce00)
	/usr/local/go/src/net/http/httputil/reverseproxy.go:278 +0x3e7
github.com/containous/traefik/v2/pkg/middlewares/pipelining.(*pipelining).ServeHTTP(0xc0005cf240, 0x2e2a100, 0xc0005cc050, 0xc0000fce00)
	/go/src/github.com/containous/traefik/pkg/middlewares/pipelining/pipelining.go:36 +0xe7
github.com/containous/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc000263bc0, 0x2e2a100, 0xc0005cc050, 0xc0000fce00)
	/go/src/github.com/containous/traefik/pkg/middlewares/accesslog/field_middleware.go:29 +0x18d
github.com/vulcand/oxy/roundrobin.(*RoundRobin).ServeHTTP(0xc0000235e0, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)
	/go/pkg/mod/github.com/vulcand/oxy@v1.1.0/roundrobin/rr.go:147 +0x13d
github.com/containous/traefik/v2/pkg/middlewares/emptybackendhandler.(*emptyBackend).ServeHTTP(0xc0005cf270, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)
	/go/src/github.com/containous/traefik/pkg/middlewares/emptybackendhandler/empty_backend_handler.go:31 +0x168
github.com/containous/traefik/v2/pkg/middlewares/tracing.(*forwarderMiddleware).ServeHTTP(0xc0002e0d20, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)
	/go/src/github.com/containous/traefik/pkg/middlewares/tracing/forwarder.go:38 +0x609
github.com/containous/traefik/v2/pkg/middlewares/ipwhitelist.(*ipWhiteLister).ServeHTTP(0xc000263c80, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)
	/go/src/github.com/containous/traefik/pkg/middlewares/ipwhitelist/ip_whitelist.go:76 +0x3da
github.com/containous/traefik/v2/pkg/middlewares/tracing.(*Wrapper).ServeHTTP(0xc0002e18f0, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)
	/go/src/github.com/containous/traefik/pkg/middlewares/tracing/wrapper.go:57 +0x16a
github.com/containous/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc000263cc0, 0x2e2a100, 0xc0005cc050, 0xc0000fcd00)
	/go/src/github.com/containous/traefik/pkg/middlewares/accesslog/field_middleware.go:29 +0x18d
github.com/gorilla/mux.(*Router).ServeHTTP(0xc00007fc20, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)
	/go/pkg/mod/github.com/containous/mux@v0.0.0-20181024131434-c33f32e26898/mux.go:133 +0x112
github.com/containous/traefik/v2/pkg/middlewares/recovery.(*recovery).ServeHTTP(0xc00053ca00, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)
	/go/src/github.com/containous/traefik/pkg/middlewares/recovery/recovery.go:33 +0x104
github.com/containous/traefik/v2/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc000263ec0, 0x2e2a100, 0xc0005cc050, 0xc0000fca00)
	/go/src/github.com/containous/traefik/pkg/middlewares/accesslog/field_middleware.go:29 +0x18d
github.com/containous/traefik/v2/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0xc00011cdd8, 0x2e2a100, 0xc0005cc050, 0xc0000fc800, 0xc0005f8c70)
	/go/src/github.com/containous/traefik/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x37d
github.com/containous/traefik/v2/pkg/middlewares/requestdecorator.WrapHandler.func1.1(0x2e2a100, 0xc0005cc050, 0xc0000fc800)
	/go/src/github.com/containous/traefik/pkg/middlewares/requestdecorator/request_decorator.go:84 +0x83
net/http.HandlerFunc.ServeHTTP(0xc00053cae0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)
	/usr/local/go/src/net/http/server.go:2012 +0x44
github.com/containous/traefik/v2/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0xc00011cdc0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)
	/go/src/github.com/containous/traefik/pkg/middlewares/handler_switcher.go:23+0x70
github.com/containous/traefik/v2/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0xc00013f680, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)
	/go/src/github.com/containous/traefik/pkg/middlewares/forwardedheaders/forwarded_header.go:173 +0x10b
net/http.serverHandler.ServeHTTP(0xc0002241c0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)
	/usr/local/go/src/net/http/server.go:2807 +0xa3
net/http.initALPNRequest.ServeHTTP(0x2e33c40, 0xc0004818f0, 0xc0000f1500, 0xc0002241c0, 0x2e2a100, 0xc0005cc050, 0xc0000fc800)
	/usr/local/go/src/net/http/server.go:3381 +0x8d
net/http.(*http2serverConn).runHandler(0xc000502780, 0xc0005cc050, 0xc0000fc800, 0xc000188040)
	/usr/local/go/src/net/http/h2_bundle.go:5720 +0x8b
created by net/http.(*http2serverConn).processHeaders
	/usr/local/go/src/net/http/h2_bundle.go:5454 +0x4e1
```

It seems that it only happens when IP Whitelisting is enabled. With Traefik v.2.2.2 and IP Whitelisting disabled it is working fine.

**Comment by SantoDE** [comment]
Good morning @RogerSik,

can you please provide me with an example dynamic configuration so we can try to reproduce?

Thanks!

**Comment by RogerSik** [comment]
Good Morning @SantoDE 

here are configuration of the docker-compose.yml 
https://gist.github.com/RogerSik/232d53577f989b91f6b8f312065af13f

Not needed informations removed

**Comment by RogerSik** [comment]
config.yml https://gist.github.com/RogerSik/120581d58fe7c9e5ca6dd1d77d61c84d

traefik.yml https://gist.github.com/RogerSik/527d876a2476d980fc6de2000fc7ad17

**Comment by SantoDE** [comment]
Hello @RogerSik,

thanks for the information. We confirm the bug and will fix it ASAP. Sorry for the inconvenience it may cause.

**Comment by RogerSik** [comment]
@SantoDE nice awesome for the fast reaction 👍

**Comment by SantoDE** [comment]
Hello @RogerSik,

we've raised a PR for that P0. If you want to test an experimental image before, you can try with `juliens/traefik:fix`. Other then that, we plan to release a fixed version throughout the day.

**Comment by RogerSik** [comment]
@SantoDE can confirm with juliens/traefik:fix it's working fine. :)

**Comment by traefiker** [comment]
Closed by #7016.
